core/src/main/java/io/confluent/kafka/schemaregistry/rest/RequestHeaderHandler.java

```diff
@@ -71,6 +73,15 @@ protected void addXForwardedForToRequest(MutableHttpServletRequest mutableReques
     }
   }
 
+  protected void addXForwardedPortToRequest(MutableHttpServletRequest mutableRequest,
+                                            HttpServletRequest request) {
+    // Do not propagate on leader call, or it would override follower port
+    if (mutableRequest.getHeader(X_FORWARD_HEADER) == null
+        && mutableRequest.getHeader(X_FORWARDED_PORT_HEADER) == null) {
+      mutableRequest.putHeader(X_FORWARDED_PORT_HEADER, String.valueOf(request.getLocalPort()));
```

Comment:
I don't think we should be adding this code in schema-registry repo. Can't we add it as part of confluent-cloud-plugins only as this is cloud specific logic?

Secondly we would need to extract the port based on PrivateLinkPortExtraction handler. getLocalPort() will not return the right port for Private traffic.

We should also add a filter to scrub this in all paths except internal port in confluent cloud plugins repo.

@rayokota  thoughts?